src/spotify_to_tidal/sync.py

```diff
@@ -155,24 +155,45 @@ async def repeat_on_request_error(function, *args, remaining=5, **kwargs):
         time.sleep(sleep_schedule.get(remaining, 1))
         return await repeat_on_request_error(function, *args, remaining=remaining-1, **kwargs)
 
-async def get_tracks_from_spotify_playlist(spotify_session: spotipy.Spotify, spotify_playlist):
-    def _get_tracks_from_spotify_playlist(offset: int, spotify_session: spotipy.Spotify, playlist_id: str):
-        fields="next,total,limit,items(track(name,album(name,artists),artists,track_number,duration_ms,id,external_ids(isrc)))"
-        return spotify_session.playlist_tracks(playlist_id, fields, offset=offset)
 
+async def _fetch_all_from_spotify_in_chunks(spotify_session: spotipy.Spotify, fetch_function: Callable, reverse: bool = False) -> List[dict]:
```

Comment:
I think `spotify_session` is unused? And `reverse` doesn't really belong here imo, try to stick to the Single-responsibility principle where possible

Reply:
`spotify_session` is used by the fetch functions `_get_tracks_from_spotify_playlist` and `_get_favorite_tracks`  , but the reverse functionality does not make sense here.

Reply:
You can use a closure for the spotify session
[closure.patch](https://github.com/user-attachments/files/15745460/closure.patch)